bootstrap.py

```diff
@@ -45,206 +51,123 @@
 parser.add_argument('--gui', default=None,
                     help="GUI toolkit: pyqt5 (for PyQt5) or pyside2 "
                     "(for PySide2)")
-parser.add_argument('--show-console', action='store_true', default=False,
-                    help="(Deprecated) Does nothing, now the default behavior "
-                    "is to show the console")
 parser.add_argument('--hide-console', action='store_true', default=False,
                     help="Hide parent console window (Windows only)")
 parser.add_argument('--safe-mode', dest="safe_mode",
                     action='store_true', default=False,
                     help="Start Spyder with a clean configuration directory")
-parser.add_argument('--no-apport', action='store_true', default=False,
-                    help="Disable Apport exception hook (Ubuntu)")
 parser.add_argument('--debug', action='store_true',
                     default=False, help="Run Spyder in debug mode")
 parser.add_argument('--filter-log', default='',
                     help="Comma-separated module name hierarchies whose log "
                          "messages should be shown. e.g., "
                          "spyder.plugins.completion,spyder.plugins.editor")
+parser.add_argument('--no-install', action='store_true', default=False,
+                    help="Do not install spyder or subrepos")
 parser.add_argument('spyder_options', nargs='*')
 
 args = parser.parse_args()
 
-# Store variable to be used in self.restart (restart spyder instance)
-os.environ['SPYDER_BOOTSTRAP_ARGS'] = str(sys.argv[1:])
-
 assert args.gui in (None, 'pyqt5', 'pyside2'), \
        "Invalid GUI toolkit option '%s'" % args.gui
 
-# Start Spyder with a clean configuration directory for testing purposes
-if args.safe_mode:
-    os.environ['SPYDER_SAFE_MODE'] = 'True'
-
 # Prepare arguments for Spyder's main script
 sys.argv = [sys.argv[0]] + args.spyder_options
```

Comment:
```suggestion
script_argv = sys.argv.copy()
sys.argv = [sys.argv[0]] + args.spyder_options
```

So we have the original argv to use on a re-run